File: OCH_Tao/ABC/B/366.js
let [, ...S] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
S.reverse();
const X = Math.max(...S.map(x => x.length));
S = S.map(x => x.padEnd(X, "*"));
for (let i = 0; i < X; i++) {
  console.log(S.map(x => x[i]).join("").replace(/\*+$/, ""));
}

File: OCH_Tao/ABC/B/366.py
N = int(input())
S = []
for i in range(N):
  S.append(input())
S.reverse()
X=max(map(len,S))
for i in range(N):
  if len(S[i])<X:
    S[i]+="*"*(X-len(S[i]))
for i in range(X):
  print("".join([j[i] for j in S]).rstrip("*"))